packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -341,7 +343,14 @@ - (int64_t)position {
   return FLTCMTimeToMillis([_player currentTime]);
 }
 
+- (bool)isDurationIndefinite {
+  return CMTIME_IS_INDEFINITE([[_player currentItem] duration]);
+}
+
 - (int64_t)duration {
+  // When CMTIME_IS_INDEFINITE return a value that matches TIME_UNSET from ExoPlayer2 on Android.
```

Comment:
What's the significants to make it matches android? It seems like a high maintenance code as we don't have a test to cover it.

Reply:
The problem here to me really stems from: https://github.com/flutter/plugins/blob/07cf89a08b8800f3337049654bc5c1c19edd50d0/packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m#L14

The conversion to Milliseconds from CMTime removes some of the functionality that CMTime provides. As far as I can tell `FLTCMTimeToMillis` is already an attempt to mimic the timebase found in Exoplayer2 to make it easier to digest on the flutter side. But it is an incomplete implementation since the timebase used in Exoplayer2 uses constants to identify special cases. On iOS these special cases (CMTIME_IS_INDEFINITE being one of them) are part of the CMTime class.

I think there has to be a way to bubble up the complexity of CMTime to the flutter side and matching Exoplayer2's timebase for now is the best way to handle that. This way at least within the flutter code everything matches.

Reply:
I see, in this case, can we refactor the `if ([self isDurationIndefinite]) ` into `FLTCMTimeToMillis`?
```
int64_t FLTCMTimeToMillis(CMTime time) {
  if (CMTIME_IS_INDEFINITE(time)) { return TIME_UNSET; }
  if (time.timescale == 0) { return 0; }
  return time.value * 1000 / time.timescale;
}
```

And I think `FLTCMTimeToMillis` should really be a static function as it is only accessed in this file.

Reply:
I've done the refactor but I'm a bit at a loss at how to create a test for this. Should I create a XCTest to test passing variations of CMTime to `FLTCMTimeToMillis` and checking their output? Or should I make an integration test - in which case wouldn't I need an example HLS video hosted somewhere like https://flutter.github.io/assets-for-api-docs/assets/videos/?

Reply:
added a test